### ISSUE TITLE
EPE-13	Review design of Knowledge Base index and articles

### ISSUE THREAD

**Issue by seagrinch** [ooiepe/epe_modules#124]
**Notes from 4/23/15** 

Knowledge Base Updates
* Update main title to “Help and How To”
* Add breadcrumbs to the detail page to include the category the article is in as well as the current article.
** Help > Visualizations > Article
* clicking the category name will take the user to a list of articles in that category
** page will include the title of “Help and How To”
** also breadcrumbs
* clicking on the help item returns the user to the main help page
* Change common issues section to “Getting Started?”
* add a right rail to the detail page that incudes te search and gettign started?
* update the steps to show that the 4 images are directly associated with 4 specific steps and pushes the next step down as needed
* remove the “EPE Knowledge Base” title from the detail page and include breadcrumbs above the page title

**Notes from 1/22/15**
Key features of GitHub to replicate:
- All articles listed by category on the main kb page
- Article order is manually controlled in each category
- Article page allows you to see a breadcrumb with a link back to the kb home and a separate link to the category page that shows the articles for that category

Other notes from discussion:
- Need the ability to manually order the pages in each section. Joe: expect this to be done with a menu.
- Possibly have a “common questions” section. As opposed to github’s mix of faq style questions and “regular” articles.
- Tried the image upload capability. Worked but need to look at the image cropping sizes. Can we use a width only one without cropping.

**Comment by seagrinch** [comment]
See ooi_KnowledgeBase_list_01.jpg and ooi_KnowledgeBase_detail_01.jpg

**Comment by seagrinch** [comment]
**From Allan on 6/10/16**
Ok, this is not very obvious I have to hunt down the code to finally figure out. the kb draggable UI is on each section's landing page
/help/cm
/help/llb
/help/ev
/help/db
/help/wp

**Comment by seagrinch** [comment]
Looks good.  I don’t think there is much to implement here, except some CSS, unless we can also figure out how to get screenshots to align with the steps.

![ooi_knowledgebase_list_01](https://cloud.githubusercontent.com/assets/1587785/19903022/741497f0-a043-11e6-9600-3759e1e787d1.jpg)
![ooi_knowledgebase_detail_01](https://cloud.githubusercontent.com/assets/1587785/19903021/74125940-a043-11e6-80b0-444dc8ecefab.jpg)

**Comment by acyu** [comment]
layout change commit: https://github.com/ooiepe/epe_modules/pull/163

**Comment by seagrinch** [comment]
There are a number of remaining issues...

1. First off, while the code worked on my local machine, on our test server I got a syntax error in sites/all/modules/epe_modules/epe_db/epe_db.module on line 128.  My guess is that line 139 would also fail.  I think this is because we're running PHP 5.3, which I don't think supports the [0] syntax in if(explode('/',$nodeurl)[0] == 'help') {
2. We should remove the body text on the section index pages.  They can just be the list of titles.
3. How do we update the KB title? (see above)
4. Can we add breadcrumbs? (see above)
5. Can we rename the "Common Issues" section? (see above)
6. Also, how do we tag items to be included in the Common Issues / Getting Started section?
7. We need to remove the limit of items that are displayed on the main index and section index pages, i.e. all pages should be listed.
8. Styling needs: section headers, vertical line on index page, page titles, subtitles
9.  I also receive the following errors on the main KB index page or any of the section pages. 

> Warning: Invalid argument supplied for foreach() in element_children() (line 6541 of /Users/sage/Sites/epe-r3/includes/common.inc).
> Warning: Illegal string offset '#children' in drupal_render() (line 5998 of /Users/sage/Sites/epe-r3/includes/common.inc).
> Warning: Illegal string offset '#children' in drupal_render() (line 6008 of /Users/sage/Sites/epe-r3/includes/common.inc).
> Warning: Illegal string offset '#children' in drupal_render() (line 6046 of /Users/sage/Sites/epe-r3/includes/common.inc).
> Warning: Illegal string offset '#printed' in drupal_render() (line 6053 of /Users/sage/Sites/epe-r3/includes/common.inc).

**Comment by seagrinch** [comment]
Still pending...
* I think we still need to fix 2 above.  The body text for subpages in a section is still showing up on the section page.  It is not showing up on the KB index or about pages, which is good.
* For 5, the common issues link goes to /help, so I can't figure out how to rename it.  Shouldn't it be its own page so we can just list those items?
* For 9, I'm still getting the errors, guess we'll have to see what it looks like on dev.

**Comment by acyu** [comment]
#1, are you referring to section text such as "Placeholder content for the LLB Intro page"?  If so, I intentionally left it there because that's actually node body text, and I prefer to edit the page through UI to remove it.

#2, the help is the default kb alias that we generated, and it will show all kb page that's tag as kb.  is common issue going to have a different tag?

**Comment by seagrinch** [comment]
* If you click on a subsection title, the page you get includes the body text for all articles in that section.  It should only include the titles, like on the KB homepage and About pages.  An excerpt might be okay, but I'm not sure if Drupal supports that as a separate field like WordPress does.
* The "Common Issues" section can use the kb tag, but the problems is the Section Title links to /help which is the url for the KB as a whole, that is, you can't get to the Common Issues index.  Also, how do we rename this section, e.g. to Getting Started?

**Comment by acyu** [comment]
I think i got this confused.  so we are talking about creating a subpage for common issues, like /help/cm is for concept map, and /help/kb for common issue?

**Comment by seagrinch** [comment]
Yes, it would be nice to flag a few issues for that category.  Otherwise, I'm worried it would just be a random listing, which is why it would be better to have a way to control it.

**Comment by seagrinch** [comment]
The Common Issues list on the main KB page still lists all kb items, instead of just the ci ones.  Also, the section index pages seem to use the same column width as the main index with is two columns.  Can they be made full width?

**Comment by acyu** [comment]
Can you give me a screenshot?  because the section index page does stretch to full width

**Comment by seagrinch** [comment]
It appears to only be the case when you're logged in (or at least when an admin is logged in).  

Can you make the section headings match their respective page titles, or is that too tricky?

**Comment by acyu** [comment]
section heading is essentially page title, I remove the hard coded title, and instead will render what user put in the page title field instead.  So, just have to edit the page title of each section page.

the section title on the index page however is still hard coded.
